lib/matplotlib/sphinxext/figmpl_directive.py

```diff
@@ -0,0 +1,294 @@
+    # ../../../build/html/plot_directive/users/explain/artists/index-1.png
+    if node['srcset']:
+        srcset = _parse_srcsetNodes(node['srcset'])
+    else:
+        srcset = None
+
+    # the rst file's location:  eg /Users/username/matplotlib/doc/users/explain/artists
+    docsource = PurePosixPath(self.document['source']).parent
+
+    # get the relpath relative to root:
+    srctop = self.builder.srcdir
+    rel = relpath(docsource, srctop).replace('.', '').replace('/', '-')
+    if len(rel):
+        rel += '-'
+    # eg: users/explain/artists
+
+    imagedir = PurePosixPath(self.builder.imagedir)
+    imagedir = PurePosixPath(self.builder.outdir) / imagedir
+    # eg: /Users/username/matplotlib/doc/build/html/_images/users/explain/artists
+
+    Path(imagedir).mkdir(parents=True, exist_ok=True)
+
+    # copy all the sources to the imagedir:
+    if srcset:
+        for mult in srcset:
+            # the entries in srcset are relative to docsource's directory
+            abspath = PurePosixPath(docsource, srcset[mult])
+            name = rel + abspath.name
+            shutil.copyfile(abspath, imagedir / name)
+    else:
+        abspath = PurePosixPath(docsource, node['uri'])
+        name = rel + abspath.name
+        shutil.copyfile(abspath, imagedir / name)
+
+    return imagedir, srcset, rel
+
+
+def visit_figmpl_html(self, node):
+
+    imagedir, srcset, rel = _copy_images_figmpl(self, node)
+
+    # /doc/examples/subd/plot_1.rst
+    docsource = PurePosixPath(self.document['source'])
+    # /doc/
+    # make sure to add the trailing slash:
+    srctop = PurePosixPath(self.builder.srcdir, '')
+    # examples/subd/plot_1.rst
+    relsource = relpath(docsource, srctop)
+    # /doc/build/html
+    desttop = PurePosixPath(self.builder.outdir, '')
+    # /doc/build/html/examples/subd
+    dest = desttop / relsource
+
+    # ../../_images/ for dirhtml and ../_images/ for html
+    imagerel = relpath(imagedir, os.path.dirname(dest))
+    if self.builder.name == "dirhtml":
+        imagerel = '..' + imagerel + ''
+    else:  # html
+        imagerel = imagerel + ''
+
+    # make uri also be relative...
+    nm = PurePosixPath(node['uri'][1:]).name
+    uri = imagerel + '/' + rel + nm
+
+    # make srcset str.  Need to change all the prefixes!
+    if srcset:
+        srcsetst = ''
+        maxmult = -1
+        for mult in srcset:
+            nm = PurePosixPath(srcset[mult][1:]).name
+            # ../../_images/plot_1_2_0x.png
+            path = imagerel + '/' + rel + nm
+            srcsetst += f'{path}'
+            if mult == 0:
+                srcsetst += ', '
+            else:
+                srcsetst += f' {mult:1.2f}x, '
+
+            if mult > maxmult:
+                maxmult = mult
+                maxsrc = path
+
+        # trim trailing comma and space...
+        srcsetst = srcsetst[:-2]
+    else:
+        srcsetst = ''
+        maxsrc = uri
```

Comment:
```suggestion
        srcsetst = srcsetst[:-2]
```

Reply:
I think its true that as we are using it srcset is always present, but in general it may not be...